storm-client/src/jvm/org/apache/storm/utils/ConfigUtils.java

```diff
@@ -313,6 +315,34 @@ public static File getWorkerDirFromRoot(String logRoot, String id, Integer port)
         return new File((logRoot + FILE_SEPARATOR + id + FILE_SEPARATOR + port));
     }
 
+    /**
+     * Get the given config value as a String, if possible.
+     * @param name - the config key
+     * @param conf - the config map
+     * @return - the config value converted to a String if found, otherwise null.
+     * @throws IllegalArgumentException if conf is null
+     * @throws NullPointerException if name is null and the conf map doesn't support null keys
+     */
+    public static String getConfigValueAsString(String name, Map<?, ?> conf) {
```

Comment:
If we are going to a common type for *childopts it should be a `List<String>` not `String`.  The reason for a list is that something with whitespace in it can be properly represented this breaks that.